packages/calendar/src/utils/stateful/config/calendar-config.impl.ts

```diff
@@ -15,39 +15,46 @@ import { DayBoundariesInternal } from '@schedule-x/shared/src/types/calendar/day
 import { DEFAULT_DAY_BOUNDARIES } from '../../../constants'
 import { timePointsPerDay } from '@schedule-x/shared/src/utils/stateless/time/time-points/time-points-per-day'
 import { Signal, signal } from '@preact/signals'
+import { CustomLocale } from '@schedule-x/shared/src/types/customLocale'
+import { CalendarCallbacks } from '@schedule-x/shared/src/interfaces/calendar/listeners.interface'
 
 export default class CalendarConfigImpl implements CalendarConfigInternal {
   firstDayOfWeek: Signal<WeekDay>
   views: Signal<View[]>
   dayBoundaries: Signal<DayBoundariesInternal>
   weekOptions: Signal<WeekOptions>
-  calendars: Signal<Record<string, CalendarType>>
-  isDark: Signal<boolean>
+  // @ts-expect-error why
+  calendars: Signal<Record<string, CalendarType> | undefined>
+  // @ts-expect-error why
+  isDark: Signal<boolean | undefined>
   minDate: Signal<string | undefined>
   maxDate: Signal<string | undefined>
-  monthGridOptions: Signal<MonthGridOptions>
+  // @ts-expect-error why
+  monthGridOptions: Signal<MonthGridOptions | undefined>
   locale: Signal<string> = signal(DEFAULT_LOCALE)
+  // @ts-expect-error nothing
+  customLocale: Signal<CustomLocale | undefined>
 
   constructor(
     locale: string = DEFAULT_LOCALE,
+    customLocale: CustomLocale | undefined,
     firstDayOfWeek: WeekDay = DEFAULT_FIRST_DAY_OF_WEEK,
     public defaultView: ViewName = InternalViewName.Week,
     views: View[] = [],
     dayBoundaries: DayBoundariesInternal = DEFAULT_DAY_BOUNDARIES,
     weekOptions: WeekOptions,
-    calendars = {},
+    calendars?: Record<string, CalendarType> | undefined,
     public plugins = {},
-    isDark: boolean = false,
-    public isResponsive: boolean = true,
-    public callbacks = {},
+    isDark?: boolean | undefined,
+    public isResponsive?: boolean | undefined,
+    public callbacks?: CalendarCallbacks | undefined,
     public _customComponentFns = {},
     minDate: string | undefined = undefined,
     maxDate: string | undefined = undefined,
-    monthGridOptions: MonthGridOptions = {
-      nEventsPerDay: 4,
-    }
+    monthGridOptions?: MonthGridOptions | undefined
   ) {
     this.locale = signal(locale)
+    this.customLocale = signal(customLocale)
```

Comment:
I like the approach, because now it's a reactive signal, so it can be updated at any time as well if we add a method for updating translations in the calendar-controls plugin.

But I think possibly even better, is if we just change `customLocale` into `languages` instead, and not make any difference between custom or languages from the library.

Then we can:
1) make it possible to add all languages, default or custom, through the config object, _or_
2) use the default translations as a fallback, if the implementer doesn't configure translations

gives us even more flexibility, and a bit less complexity in the translator function.